Create Cloud_Connection_Issues.md (#2666)
SkylineCommunications/dataminer-docs: user-guide/Troubleshooting/Procedures/Cloud_Connection_Issues.md

---
uid: Cloud_Connection_Issues
---

# Cloud Connection Issues
This page provides solutions for issues that may arise from connectivity problems with cloud services.

## Remote Access Issues
Follow the guide below when you are encountering connection issues with dataminer.services features and your DataMiner System.

### Check Status of DxMs
A common cause for issues with [dataminer.services](xref:AboutCloudPlatform) features and your DMS is due to the connection from the [DxMs](xref:DataMinerExtensionModules) towards dataminer.services being unavailable. 
Resolve this by taking the following steps:
1. Verify whether the installed [DxMs](xref:DataMinerExtensionModules) are running, otherwise start them.
1. Make sure to have the most [recent versions](xref:DCP_change_log) of the DxMs installed to avoid dealing with issues that might have been fixed. Info about upgrading DxMs can be found [here](https://community.dataminer.services/dataminer-cloud-pack/).

### Check DMS Status

> Verify the status of the DMS and ensure that the system is operational, if not, start the DMS. 

Reviewer: Maybe here we can add something about how they can validate if the WebAPI are running (don't need to mention WebAPI) by telling them to navigate to the IP defined in MaintenanceSettings.xml and see if they can access it locally (= WebAPI works locally)

### Check Cloud Session

> Verify in DataMiner Cube if the cloud session is still available and not expired, otherwise [renew the cloud connection](#renew-cloud-connection).

Reviewer: ```suggestion
Verify in DataMiner Cube if the cloud identity is still valid, otherwise please [renew the cloud identity](#renew-cloud-connection).
```

We renamed it in Cube this week :D

### Undetermined Problem
When the previous solutions did not solve the problem, ensure that all [requirements](xref:Accessing_DMS_remotely_with_Cube) for remote access are fulfilled.

## Renew Cloud Connection

> If the cloud connection expires, a new token must be requested to renew it. To do this, access the DataMiner Cube and select *Cloud* under *System Center*. If the cloud connection is expired, a ```Renew Session``` button should be available. Manually renewing the connection can be necessary if the cloud could not renew its tokens automatically for too long (no internet access for extended periods for example, usually minimum 3 days). 

Reviewer: ```suggestion
If the cloud identity expires, a new token must be requested to renew it. To do this, access the DataMiner Cube and select *Cloud* under *System Center*. If the cloud identity is expired, a ```Renew Session``` button should be available. Manually renewing the identity can be necessary if the cloud could not renew its tokens automatically for too long (no internet access for extended periods for example, usually minimum 3 days). 
```

Reviewer 2: `a new token must be requested to just `it must be renewed`

>[!NOTE]
>The ```Renew Session``` button is only available for users with the Owner role of the DMS.

> Please contact [support](https://skyline.be/contact/tech-support) if none of the above solved the problem.

Reviewer: ```suggestion
If the issues persist, please contact [support](https://skyline.be/contact/tech-support).
```

---
uid: Cloud_Connection_Issues
---

# Cloud Connection Issues

> This page provides solutions for potential connectivity problems with cloud services.

Reviewer 2: ```suggestion
This page provides solutions for potential connectivity problems between your DMS and dataminer.services.
```

> This page provides solutions for issues that may arise from connectivity problems with cloud services.

Reviewer 3: Issues and problems is redundant

```suggestion
This page provides solutions for potential connectivity problems with dataminer.services.
```

## Remote Access Issues

> Follow the guide below when you are encountering connection issues with dataminer.services features and your DataMiner System.

Reviewer 3: This sentence doesn't say much and could be removed imo

Author: Going to leave it, it makes a subcategory for specific issues about Remote Access.

### Check Status of DxMs
A common cause for issues with [dataminer.services](xref:AboutCloudPlatform) features and your DMS is due to the connection from the [DxMs](xref:DataMinerExtensionModules) towards dataminer.services being unavailable. 
Resolve this by taking the following steps:
1. Verify whether the installed [DxMs](xref:DataMinerExtensionModules) are running, otherwise start them.

> 1. Make sure to have the most [recent versions](xref:DCP_change_log) of the DxMs installed to avoid dealing with issues that might have been fixed. Info about upgrading DxMs can be found [here](https://community.dataminer.services/dataminer-cloud-pack/).

Reviewer 2: ```suggestion
1. Make sure to have the most [recent versions](xref:DCP_change_log) of the DxMs installed to avoid dealing with issues that might have been fixed, see [Upgrading nodes to the latest DxM versions](xref:Managing_cloud-connected_nodes#upgrading-nodes-to-the-latest-dxm-versions).
```

> 1. Verify the accessibility of the DMA host by opening a browser and navigating to the address specified in the [MaintenanceSettings.xml](xref:MaintenanceSettings_xml). This address corresponds to the value that has been set as the *name* parameter in the [HTTPS setting](xref:Setting_up_HTTPS_on_a_DMA) of MaintenanceSettings.xml. If the page manages to load, it indicates that the DMA host is locally accessible but there is a possible issue with the [proxy](xref:Connect_to_cloud_via_proxy) or [DMZ](xref:Connect_to_cloud_with_DMZ) settings.

Reviewer 2: ```suggestion
1. Verify the accessibility of the DMA host by opening a browser and navigating to the address specified in the [MaintenanceSettings.xml](xref:MaintenanceSettings_xml). This address corresponds to the value that has been set as the *name* parameter in the [HTTPS setting](xref:Setting_up_HTTPS_on_a_DMA) of `MaintenanceSettings.xml`. If the page manages to load, it indicates that the DMA host is locally accessible but there is a possible issue with the [proxy](xref:Connect_to_cloud_via_proxy) or [DMZ](xref:Connect_to_cloud_with_DMZ) settings.
```

### Check DMS Status
Verify the status of the DMS and ensure that the system is operational, if not, start the DMS. 

> ### Check Cloud Session

Reviewer 2: a bit like a duplicate, merge them?

Author: No will keep it separate for if more information is added in the future

> Verify in DataMiner Cube if the cloud identity is still valid, otherwise please [renew the cloud identity](#renew-cloud-identity).

Reviewer 2: ```suggestion
Verify in DataMiner Cube if the cloud identity is still valid, otherwise [renew the cloud identity](#renew-cloud-identity).
```

> ### Undetermined Problem

Reviewer 2: put this one at the bottom?

Author: No, will keep the structure as it is, this part refers to the other paragraph.

---
uid: Cloud_Connection_Issues
---

# Investigating dataminer.services connection issues

This page provides solutions for potential connectivity problems with [dataminer.services](xref:AboutCloudPlatform).

## Remote access issues

If you encounter issues with dataminer.services features that may be caused by connection issues, follow the actions below in the specified order:

1. [Check the status of the DxMs](#check-the-status-of-the-dxms).
1. [Check the status of the DMS](#check-the-status-of-the-dms).
1. [Check the cloud session](#check-the-cloud-session)
1. [Make sure requirements are met](#make-sure-requirements-are-met)

### Check the status of the DxMs

A common cause for issues with dataminer.services features is a problem with the connection from the [DxMs](xref:DataMinerExtensionModules) towards dataminer.services.

You can resolve this as follows:

> 1. Verify whether all installed [DxMs](xref:DataMinerExtensionModules) are running, otherwise start them.

Reviewer 4: Where and how should users verify this?

1. Make sure the most recent versions of the DxMs are installed. See [Upgrading nodes to the latest DxM versions](xref:Managing_cloud-connected_nodes#upgrading-nodes-to-the-latest-dxm-versions).

   This way, you avoid dealing with issues that have already been fixed.

> 1. Verify the accessibility of the DMA host:

Reviewer 4: This doesn't really seem to be related to the status of the DxMs. I think we should move this step and its substeps to the "Check the status of the DMS" section below, as the description of how the user can check whether the DMA is running. The existing info there can then be added at the end and rephrased as `If the page does not load correctly, this indicates a problem with the DMA host. Restart the DMA to resolve this. You can do so using the [DataMiner Taskbar Utility](xref:DataMiner_Taskbar_Utility).`

   1. In the `C:\Skyline DataMiner\` folder of the DMA, open the file *MaintenanceSettings.xml*.

   1. Look up the address specified in the *name* attribute of the *HTTPS* tag. For example:

       ```xml
       <MaintenanceSettings>
          ...
          <HTTPS enabled="true" name="foo.skyline.local"/>
          ...
       </MaintenanceSettings>
       ```

   1. Open a browser and navigate to this address.

      If the page loads correctly, this indicates that the DMA host is locally accessible, but there may be an issue with the [proxy](xref:Connect_to_cloud_via_proxy) or [DMZ](xref:Connect_to_cloud_with_DMZ) settings.

### Check the status of the DMS

Verify the status of the DMS and ensure that the system is operational, if not, start the DMS.

### Check the cloud session

> In DataMiner Cube, go to *System Center* > *Cloud* and check if the cloud identity is still valid.

Reviewer 4: Why is "Renewing the cloud identity" kept in a separate section below? It would be more user-friendly if we immediately detailed this here. Also, if we do move this and remove the separate section below, is it still necessary to keep the "Remote access issues" subtitle (line 9)? Do you expect to add another category of issues later or is this the only category this page will ever contain?
